### PR TITLE
[Synapse] Add Get Workspace, Read/List BigDataPools, SqlPools, IntegrationRuntimes API spec from the data-plane

### PR DESCRIPTION
**Pull request by cool-mist**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      {
+        "$ref": "#/parameters/SynapseDnsSuffixInPath"
+      }
+    ]
+  },
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/bigDataPools": {
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json:27]: >           [](start = 22, length = 10)

remove trailing spaces

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        "required": true,
+        "type": "string",
+        "in": "path",
+        "x-ms-skip-url-encoding": true
+      },
+      {
+        "$ref": "#/parameters/SynapseDnsSuffixInPath"
+      }
+    ]
+  },
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json:22]: Please reference code [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2020-02-01-preview/roles.json#L7-L15) for this part.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json (hunk 1 above)
Comment (cool-mist) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json]: Thanks, fixed trailing whitespaces in all new files added as part of this PR.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json (hunk 2 above)
Comment (cool-mist) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json:15]: Thanks for the pointer, updated this in all the new files.

File: specification/synapse/data-plane/readme.md [hunk 3]
```diff
@@ -65,6 +65,18 @@ input-file:
 - Microsoft.Synapse/preview/2019-06-01-preview/artifacts.json
 ```
 
+### Tag: package-arm-artifacts-2020-01-01-preview
+
+These settings apply only when `--tag=package-access-control-2020-01-01-preview` is specified on the command line.
```
Comment (idear1203) [inline specification/synapse/data-plane/readme.md:70]: >package-access-control-2020-01-01-preview [](start = 38, length = 41)

Let's use `package-arm-artifacts-2020-01-01-preview`. I need to confirm with PM side about the API version string and package name

File: specification/synapse/data-plane/readme.md [hunk 4]
```diff
@@ -65,6 +65,18 @@ input-file:
...
+
+``` yaml $(tag) == 'package-access-control-2020-01-01-preview'
```
Comment (idear1203) [inline specification/synapse/data-plane/readme.md:72]: >package-access-control-2020-01-01-preview [](start = 20, length = 41)

package-arm-artifacts-2020-01-01-preview

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json [hunk 5]
```diff
@@ -0,0 +1,189 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-01-01-preview",
+    "title": "SynapseClient"
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json:5]: >SynapseClient [](start = 14, length = 13)

We got some comments from SDK team before. Let's name this as `BigDataPoolClient`

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/integrationRuntimes.json [hunk 6]
```diff
@@ -0,0 +1,127 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-01-01-preview",
+    "title": "SynapseClient"
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/integrationRuntimes.json:5]: >SynapseClient [](start = 14, length = 13)

Let's name this as `IntegrationRuntimeClient`

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/sqlPools.json [hunk 7]
```diff
@@ -0,0 +1,186 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-01-01-preview",
+    "title": "SynapseClient"
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/sqlPools.json:5]: >SynapseClient [](start = 14, length = 13)

Let's name this as `SqlPoolClient`

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/workspace.json [hunk 8]
```diff
@@ -0,0 +1,170 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-01-01-preview",
+    "title": "SynapseClient"
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/workspace.json:5]: >SynapseClient [](start = 14, length = 13)

Let's name this as `WorkspaceClient`

**Comment by cool-mist** [comment]
Waiting on PM approval on the package name and api-version

File: specification/synapse/data-plane/readme.md [hunk 9]
```diff
@@ -65,6 +65,18 @@ input-file:
...
+input-file:
+- Microsoft.Synapse/preview/2020-01-01-preview/workspace.json
+- Microsoft.Synapse/preview/2020-01-01-preview/sqlPools.json
+- Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json
+- Microsoft.Synapse/preview/2020-01-01-preview/integrationRuntimes.json
+```
```
Comment (idear1203) [inline specification/synapse/data-plane/readme.md:78]:
``` yaml $(tag) == 'package-artifacts-2019-06-01-preview'
input-file:
- Microsoft.Synapse/preview/2019-06-01-preview/artifacts.json
- Microsoft.Synapse/preview/2019-06-01-preview/workspace.json
- Microsoft.Synapse/preview/2019-06-01-preview/sqlPools.json
- Microsoft.Synapse/preview/2019-06-01-preview/bigDataPools.json
- Microsoft.Synapse/preview/2019-06-01-preview/integrationRuntimes.json
```

File: specification/synapse/data-plane/readme.md [hunk 10]
```diff
@@ -65,6 +65,18 @@ input-file:
...
+### Tag: package-arm-artifacts-2020-01-01-preview
+
+These settings apply only when `--tag=package-access-control-2020-01-01-preview` is specified on the command line.
+
+``` yaml $(tag) == 'package-access-control-2020-01-01-preview'
+input-file:
+- Microsoft.Synapse/preview/2020-01-01-preview/workspace.json
+- Microsoft.Synapse/preview/2020-01-01-preview/sqlPools.json
+- Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json
+- Microsoft.Synapse/preview/2020-01-01-preview/integrationRuntimes.json
```
Comment (idear1203) [inline specification/synapse/data-plane/readme.md:77]: input-file:
- Microsoft.Synapse/preview/2019-06-01-preview/artifacts.json
- Microsoft.Synapse/preview/2019-06-01-preview/workspace.json
- Microsoft.Synapse/preview/2019-06-01-preview/sqlPools.json
- Microsoft.Synapse/preview/2019-06-01-preview/bigDataPools.json
- Microsoft.Synapse/preview/2019-06-01-preview/integrationRuntimes.json

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json (hunk 5 above)
Comment (cool-mist) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/bigDataPools.json]: Thanks, resolved it.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/integrationRuntimes.json (hunk 6 above)
Comment (cool-mist) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/integrationRuntimes.json]: Thanks, resolved it.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/sqlPools.json (hunk 7 above)
Comment (cool-mist) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/sqlPools.json]: Thanks, resolved it.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/workspace.json (hunk 8 above)
Comment (cool-mist) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2020-01-01-preview/workspace.json]: Thanks, resolved it.

File: specification/synapse/data-plane/readme.md (hunk 10 above)
Comment (cool-mist) [reply specification/synapse/data-plane/readme.md]: Updated these. Thanks.

**Comment by tjprescott** [comment]
You'll need to address all the failing checks labeled "Required" when you are ready for this to be merged.

**Comment by cool-mist** [comment]
@tjprescott - I have fixed the errors reported in the pipeline. I see that 'LintDiff' is still failing. I do not get that error on my local and the error that I see on the pipeline seems to be inconsistent with the current state of the code - https://dev.azure.com/azure-sdk/public/_build/results?buildId=465799&view=logs&j=688669d0-441c-57c3-cf6d-f89a22ccfa5d&t=923b5a90-e462-5994-c166-86509c6bf5f2

Is this in error ?

**Comment by tjprescott** [comment]
@cool-mist I will look into it and get back to you. From the error message it seems like it might be?

**Comment by cool-mist** [comment]
@tjprescott - Do you have any updates on this ?

**Comment by tjprescott** [comment]
@cool-mist when I look at the LintDiff errors now, they seem legitimate:

```
'(node:5599) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 104)\n' +
  "(node:5599) UnhandledPromiseRejectionWarning: Error: '$.definitions.Resource.properties.id.description' has incompatible values (---\n" +
```

Otherwise I would reach out to @lmazuel.

**Comment by tjprescott** [comment]
@cool-mist please take a look at the BreakingChange failures.

**Comment by idear1203** [comment]
@tjprescott some of scheme definitions in artifacts.json are not used but they cause conflicts. So I removed them. `SubResource` also has conflict with the one defined in management API (integrationRuntime.json) so I modified both to have same definitions. Synapse is still under preview so I suppose these breaking changes are fine.

**Comment by tjprescott** [comment]
@lmazuel will need to do the merge. I don't have the necessary permissions.

**Pull request merged**